### PR TITLE
Add docs for `c51.py` and `c51_atari.py`

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] New algorithm
- [x] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [x] I have updated the documentation and previewed the changes via `mkdocs serve`.
- [x] I have updated the tests accordingly (if applicable).

If you are adding new algorithms or your change could result in performance difference, you may need to (re-)run tracked experiments. See https://github.com/vwxyzjn/cleanrl/pull/137 as an example PR. 
- [x] I have contacted @vwxyzjn to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark) (**required**).
- [x] I have tracked applicable experiments in [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) with `--capture-video` flag toggled on (**required**).
- [x] I have added additional documentation and previewed the changes via `mkdocs serve`.
    - [x] I have explained note-worthy implementation details.
    - [x] I have explained the logged metrics.
    - [x] I have added links to the original paper and related papers (if applicable).
    - [x] I have added links to the PR related to the algorithm.
    - [x] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [x] I have added the learning curves (in PNG format with `width=500` and `height=300`).
    - [x] I have added links to the tracked experiments.
- [x] I have updated the tests accordingly (if applicable).

### CODE REVIEW

**Comment by vwxyzjn** [comment]
This branch is ready for review.

**Comment by vwxyzjn** [comment]
This branch is ready for review.

File: benchmark/c51/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Categorical DQN (C51) Benchmark
+
+This repository contains instructions to reproduce our DQN experiments.
```
Comment (dosssman) [inline benchmark/c51/README.md:3]: Categorical DQN

File: benchmark/c51/atari.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# BeamRiderNoFrameskip-v4
+poetry run python cleanrl/c51_atari.py --env-id BeamRiderNoFrameskip-v4 --track --capture-video --seed 1 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
+poetry run python cleanrl/c51_atari.py --env-id BeamRiderNoFrameskip-v4 --track --capture-video --seed 2 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
+poetry run python cleanrl/c51_atari.py --env-id BeamRiderNoFrameskip-v4 --track --capture-video --seed 3 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
+
+# BreakoutNoFrameskip-v4
+poetry run python cleanrl/c51_atari.py --env-id BreakoutNoFrameskip-v4 --track --capture-video --seed 1 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
+poetry run python cleanrl/c51_atari.py --env-id BreakoutNoFrameskip-v4 --track --capture-video --seed 2 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
+poetry run python cleanrl/c51_atari.py --env-id BreakoutNoFrameskip-v4 --track --capture-video --seed 3 --wandb-project-name cleanrl --wandb-entity openrlbenchmark
```
Comment (dosssman) [inline benchmark/c51/atari.sh:14]: Git not happy

File: cleanrl/c51.py [hunk 3]
```diff
@@ -50,21 +50,21 @@ def parse_args():
...
         help="the starting epsilon for exploration")
-    parser.add_argument("--end-e", type=float, default=0.02,
+    parser.add_argument("--end-e", type=float, default=0.05,
         help="the ending epsilon for exploration")
     parser.add_argument("--exploration-fraction", type=float, default=0.5,
         help="the fraction of `total-timesteps` it takes from start-e to go end-e")
     parser.add_argument("--learning-starts", type=int, default=10000,
         help="timestep to start learning")
-    parser.add_argument("--train-frequency", type=int, default=1,
+    parser.add_argument("--train-frequency", type=int, default=10,
```
Comment (dosssman) [inline cleanrl/c51.py:67]: Same comment as for DQN. If the Q network being update 10 times slower, does the agent still learn ?
What is the motivation for this change ?

File: docs/rl-algorithms/c51.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
+# Categorical DQN (C51)
+
+## Overview
+
+C51 introduces a distribution of perspective for DQN: instead of learning a single value for an action, C51 learns to predict a distribution of values for the action. With this perspective C51 introduces applies the Bellman's equation to learning the approximate value distribution. Empirically, C51 demonstrates impressive performance in ALE.
```
Comment (dosssman) [inline docs/rl-algorithms/c51.md:5]: - introduces a __distributional perspective__ for DQN ?

- With this perspective C51 introduces ... -> "To estimate the distribution of value for a given action,, Categorical DQN (C51) discretizes the support of the value distribution into N=51 atoms"

**Review by dosssman — COMMENTED** [review]
Hello here.
Sorry for the late review.

Beside the few single comments, it looks quite good to me.
Great job as always.

File: benchmark/c51/README.md (hunk 1 above)
Comment (vwxyzjn) [reply benchmark/c51/README.md:3]: Yeah this README.md stuff is not maintainbale. I am going to fix this now, but in the future let's do it via #165

File: cleanrl/c51.py (hunk 3 above)
Comment (vwxyzjn) [reply cleanrl/c51.py:67]: This is just a hyperparameter, see https://github.com/vwxyzjn/cleanrl/pull/157#discussion_r853678674

File: docs/rl-algorithms/c51.md (hunk 4 above)
Comment (vwxyzjn) [reply docs/rl-algorithms/c51.md]: I just simplified it...

**Comment by vwxyzjn** [comment]
Thank you! Please approve the PR if everything looks good to you :)

**Review by dosssman — APPROVED** [review]
